api/cluster.go

```diff
@@ -228,12 +215,22 @@ func CreateCluster(c *gin.Context) {
 		ResourceID: commonCluster.GetID(),
 	})
 
-	go postCreateCluster(commonCluster)
+	go postCreateCluster(commonCluster, &createClusterRequest)
```

Comment:
If we run the validation in a separate go routine than how the validation error is returned back to the caller?